presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -192,10 +197,27 @@ else if (queryStateMachine.getQueryState() == QueryState.STARTING) {
         ImmutableList.Builder<SqlStageExecution> stages = ImmutableList.builder();
 
         StageId stageId = new StageId(queryStateMachine.getQueryId(), String.valueOf(nextStageId.getAndIncrement()));
+
+        // Non-broadcast distributions declare output buffers immediately
+        OutputBuffers initialOutputBuffers = createInitialEmptyOutputBuffers();
+        if (!plan.getFragment().getPartitionFunction().getPartitioningHandle().equals(FIXED_BROADCAST_DISTRIBUTION)) {
```

Comment:
Alternatively, you could do the initial output buffer declaration inside constructor of PartitionOutputBufferManager, and do nothing in `addOutputBuffers` (except maybe sanity check)